src/core/qgsfield.h

```diff
@@ -433,6 +433,20 @@ class CORE_EXPORT QgsField
      */
     QgsEditorWidgetSetup editorWidgetSetup() const;
 
+    /**
+     * Make field read-only if \a readOnly is set to true. This is the case for
+     * providers which support generated fields for instance.
+     * \since QGIS 3.18
+     */
+    void setReadOnly( bool readOnly );
+
+    /**
+     * Returns TRUE if this field is a read-only field. This is the case for
+     * providers which support generated fields for instance.
+     * \since QGIS 3.18
+     */
+    bool isReadOnly() const;
```

Comment:
Can you add this as property too? This will make it easier to integrate it in QML based apps like qfield.

Reply:
Yes, sure. Done